src/main/java/io/xpire/model/Xpire.java

```diff
@@ -1,89 +1,126 @@
 package io.xpire.model;
 
+import static io.xpire.model.tag.Tag.EXPIRED_TAG;
 import static java.util.Objects.requireNonNull;
 
+import java.util.Iterator;
 import java.util.List;
-
-import io.xpire.model.item.Item;
-import io.xpire.model.item.SortedUniqueItemList;
-import io.xpire.model.item.sort.MethodOfSorting;
+import java.util.Set;
+import java.util.TreeSet;
+
+import io.xpire.model.item.SortedUniqueXpireItemList;
+import io.xpire.model.item.XpireItem;
+import io.xpire.model.item.sort.XpireMethodOfSorting;
+import io.xpire.model.tag.Tag;
+import io.xpire.model.tag.TagComparator;
 import javafx.collections.ObservableList;
 
 /**
  * Wraps all data at xpire level
  * Duplicates are not allowed (by .isSameItem comparison)
  */
-public class Xpire implements ReadOnlyXpire {
+public class Xpire implements ReadOnlyListView<XpireItem> {
 
-    private final SortedUniqueItemList items = new SortedUniqueItemList();
+    private final SortedUniqueXpireItemList items = new SortedUniqueXpireItemList();
 
     public Xpire() {}
 
     /**
      * Creates a Xpire object using the Items in the {@code toBeCopied}
      */
-    public Xpire(ReadOnlyXpire toBeCopied) {
+    public Xpire(ReadOnlyListView toBeCopied) {
         this();
         this.resetData(toBeCopied);
     }
 
     //// list overwrite operations
 
     /**
-     * Replaces the contents of the item list with {@code items}.
-     * {@code items} must not contain duplicate items.
+     * Replaces the contents of the xpireItem list with {@code xpireItems}.
+     * {@code xpireItems} must not contain duplicate xpireItems.
      */
-    public void setItems(List<Item> items) {
-        this.items.setItems(items);
+    public void setItems(List<XpireItem> xpireItems) {
+        this.items.setItems(xpireItems);
     }
 
     /**
      * Resets the existing data of this {@code Xpire} with {@code newData}.
      */
-    public void resetData(ReadOnlyXpire newData) {
+    public void resetData(ReadOnlyListView newData) {
         requireNonNull(newData);
         this.setItems(newData.getItemList());
     }
 
     /**
-     * Returns true if an item with the same identity as {@code item} exists in xpire.
+     * Returns true if an xpireItem with the same identity as {@code xpireItem} exists in xpire.
      */
-    public boolean hasItem(Item item) {
-        requireNonNull(item);
-        return this.items.contains(item);
+    public boolean hasItem(XpireItem xpireItem) {
+        requireNonNull(xpireItem);
+        return this.items.contains(xpireItem);
     }
 
     /**
-     * Adds a item to xpire.
-     * The item must not already exist in xpire.
+     * Adds a xpireItem to xpire.
+     * The xpireItem must not already exist in xpire.
```

Comment:
Perhaps `The xpireItem` can be replaced with `{@code xpireItem}` to match with the rest of the code :)

Reply:
Noted! :)